File: prototype/Assets/microcosmicWar/Scripts/lifeAction/destroyWhenDie.js

var delayTime=0.0;

function Start()
{
	var life:Life = gameObject.GetComponentInChildren(Life);
	life.addDieCallback(deadAction);
}


//在死亡的回调中使用
function deadAction()
{
	Destroy(gameObject,delayTime);
}

File: prototype/Assets/microcosmicWar/Scripts/Event/LifeEnd/destroyWhenDie.cs

using UnityEngine;
using System.Collections;

public class destroyWhenDie : MonoBehaviour
{


    public float delayTime = 0.0f;

    void Start()
    {
        Life life = gameObject.GetComponentInChildren<Life>();
        life.addDieCallback(deadAction);
    }


    //在死亡的回调中使用
    void deadAction(Life p)
    {
        Destroy(gameObject, delayTime);
    }
}